src/components/single-article.jsx

```diff
@@ -19,6 +19,18 @@ function SingleArticle() {
     });
   }, [article_id]);
 
+  const handleVote = () => {
+    setSingleArticle((currentArticle) => {
+      return { ...currentArticle, votes: currentArticle.votes + 1 };
+    });
+
+    patchArticleVote(singleArticle.article_id).catch(() => {
+      setSingleArticle((currentArticle) => {
+        return { ...currentArticle, votes: currentArticle.votes - 1 };
+      });
+    });
+  };
+
```

Comment:
This is fine but you could make this even better by creating a state specifically for the votes and rendering/updating that instead. That way you only have to do a set state operation on a primitive data value as opposed to a whole object which is quicker and less expensive.